server/app/routes/reviews.js

```diff
@@ -27,11 +22,35 @@ router.get('/user/:userId', function (req, res, next) {
 });
 
 router.post('/', utils.ensureAuthenticated, function (req, res, next) {
+  var productId = req.body.productId;
+  var userId = req.body.userId;
+  var review;
+
   Review.create(req.body)
   .then(function(createdReview){
-    res.status(201).json(createdReview);
+      review = createdReview;
+    return review.setUser(userId);
+  }).then(function(resultObj){
+    if(resultObj){
```

Comment:
Do you need to be checking if there is a resultObj returned? Have you experienced a situation when you are not returned anything but it hits the `.then`?